### PR TITLE
ref: separate trainer docstrings poc

### PR DESCRIPTION
**Pull request by williamFalcon**

### CODE REVIEW

**Comment by Borda** [comment]
Well, I don't like this way... Docs are to understand what a function is doing without reading code
This most likely does have support in any IDE so finding function finding describes will be super hard

**Comment by williamFalcon** [comment]
this doesn't change ide support... since ides look at the __doc__

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -103,58 +104,6 @@ class Trainer(
...
 ):
-    """
-    Example:
-
-        >>> import torch
-        >>> from torch.nn import functional as F
-        >>> from torch.utils.data import Dataset, DataLoader
-
-        >>> # Define model
-        >>> class SimpleModel(LightningModule):
```
Comment (Borda) [inline pytorch_lightning/trainer/trainer.py:114]: doctest...

**Comment by awaelchli** [comment]
> this doesn't change ide support... since ides look at the **doc**

This statement is false. IDEs parse the files directly, they don't run any imports.
After this PR, inspecting the Trainer class will not render any docstring.

**Comment by Borda** [comment]
@PyTorchLightning/core-contributors pls share your thought on this new documentation way... 🐰

**Comment by williamFalcon** [comment]
yeah, i noticed that too. The original docs about this were a bit misleading. 
anyhow, i think we can still get the IDEs to pick up on the docstrings.

**Comment by Borda** [comment]
> > this doesn't change ide support... since ides look at the **doc**
> 
> This statement is false. IDEs parse the files directly, they don't run any imports.
> After this PR, inspecting the Trainer class will not render any docstring.

this is my case in Pycharm
Screenshot 2020-09-09 at 15 10 15
Screenshot 2020-09-09 at 15 11 39

**Comment by williamFalcon** [comment]
the pros:

- clean and docs are separate... many methods have 20 lines of docs and 1 line of code. this seems silly to me and makes code unreadable. 

cons
- extra step for people to remember when adding docs.

**Comment by awaelchli** [comment]
If the problem is that the docstring is too long, I think an option is to shrink it, document only the essential, and move all tutorial/best practice notes and instructions to a matching .rst file. A nice example of this is the metrics or logger package, where we have the lengthy introductions written in the .rst files and simply include the docstrings with autoclass/autosummary directives.

**Comment by Borda** [comment]
> * code and docs are separate... many methods have 20 lines of docs and 1 line of code. this seems silly to me and makes code unreadable.

I say your case with just calling the parent method inside, in such case, we can have a simple docstring just referring to the parent docs...
Just afraid that maintaining this dislocated docs from code will be much harder, more work with PRs and great source or not updated docs according to code changes... which was pretty much why we moved from MKDocs to Sphinx

**Comment by Borda** [comment]
@awaelchli I guess that also this extra strings does not work with doctest nor sphinx docs testing, right?

**Comment by Borda** [comment]
How about keeping minimalistic docstrings and all longer comenst move rst files so we have clear doc and code separation... This docstrings.py is just strange midway hack =/

**Pull request merged**